go-controller/pkg/kube/annotator.go

```diff
@@ -81,73 +81,63 @@ func (na *nodeAnnotator) Run() error {
 }
 
 // NewPodAnnotator returns a new annotator for Pod objects
-func NewPodAnnotator(kube Interface, pod *kapi.Pod) Annotator {
+func NewPodAnnotator(kube Interface, podName string, namespace string) Annotator {
 	return &podAnnotator{
-		kube:    kube,
-		pod:     pod,
-		changes: make(map[string]*action),
+		kube:      kube,
+		podName:   podName,
+		namespace: namespace,
+		changes:   make(map[string]interface{}),
 	}
 }
 
 type podAnnotator struct {
-	kube Interface
-	pod  *kapi.Pod
+	kube      Interface
+	podName   string
+	namespace string
 
-	changes map[string]*action
```

Comment:
Q: it looks like all 3 flavors of annotator have these:

```
changes map[string]interface{}
sync.Mutex
```

And it also looks like other than 'Run' all of them have the same Set and Delete functions.
Do you see any value in combining there into the Annotator interface, or perhaps create
another interface that uses Annotator and has all these common elements?

PS: **This is just a suggestion!!!** Please feel free to ignore. :)

Reply:
That's a good suggestion!
I wanted to keep it as it was, for now, to avoid changing too much of the code, this PR is already quite big.
I think this can be addressed separately.

Reply:
ack. In that case,
/lgtm
;)